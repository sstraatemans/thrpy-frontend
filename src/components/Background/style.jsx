import styled from "styled-components";

export const BackgroundWrapper = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: url(${props => props.img}) no-repeat center center fixed;
  background-size: cover;
`;
