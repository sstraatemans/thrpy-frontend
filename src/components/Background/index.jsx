//@flow
import React, { Component } from "react";
import { BackgroundWrapper } from "./style";

type Props = {};
type State = {
  img: number
};
class Background extends Component<Props, State> {
  getImage: Function;
  setImage: Function;

  constructor(props: Props) {
    super(props);

    this.state = {
      img: 5
    };

    this.getImage = this.getImage.bind(this);
    this.setImage = this.setImage.bind(this);
  }

  componentWillMount() {
    this.setImage();
  }

  setImage(): void {
    let numb: number = parseInt(sessionStorage.getItem("backimage"));
    if (!numb) {
      numb = Math.round(Math.random() * 22);
      this.setState({ img: numb });
      sessionStorage.setItem("backimage", `${numb}`);
    }

    this.setState({ img: numb });
  }
  getImage(): string {
    return `images/back/${this.state.img}.jpg`;
  }
  render() {
    return <BackgroundWrapper img={this.getImage()} />;
  }
}

export default Background;
