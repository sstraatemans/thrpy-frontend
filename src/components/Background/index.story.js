import React from "react";

import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import Background from "./";

storiesOf("Background", module).add("complete", () => <Background />);
