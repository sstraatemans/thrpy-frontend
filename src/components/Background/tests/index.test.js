import React from "react";
import Background from "./../";

describe("<Background />", () => {
  it("render component <Background>", () => {
    let wrapper = mount(<Background />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
