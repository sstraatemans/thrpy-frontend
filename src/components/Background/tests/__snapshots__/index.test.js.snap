exports[`<Background /> render component <Background> 1`] = `
<Background>
  <styled.div
    img="images/back/11.jpg">
    <div
      className="sc-bdVaJa ejQozR" />
  </styled.div>
</Background>
`;
